### ISSUE TITLE
flutter_news_template - packages:news_blocks_ui Error: dart(argument_type_not_assignable) and version bump to intl: ^0.18.0

### ISSUE THREAD

**Issue by TylerByte666** [flutter/flutter#127015]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

1. Install the [Flutter News Template](https://flutter.github.io/news_toolkit/).
1.1 Run ```mason add -g flutter_news_template```.  
2. Generate the app:
2.1 Run ```mason make flutter_news_template```.
3. Set Template configuration:
3.1 What is the user-facing application name? (Daily Globe)
3.2 What is the application bundle identifier? (com.flutter.news.template)
3.3 Who are the code owners? (separated by spaces) (@TylerByte666)
3.4 What flavors do you want your application to include?
 ❯ ◉  development
    ◯  integration
    ◯  staging
     ◉  production
4. Error shows up in **VS Code** dart analyzer for file packages:new_blocks_ui/lib/src/html.dart

```dart
The argument type 'void Function(String?, Map<String, String>, Element?, dynamic)' can't be assigned to the parameter type 'void Function(String?, Map<String, String>, Element?)?'
```

This is related to the import 'package:url_launcher/url_launcher.dart' being used on line 28

```dart:
onLinkTap: (url, context, attributes, element) {
  if (url == null) return;
  final uri = Uri.tryParse(url);
  if (uri == null) return;
  launchUrl(uri).ignore();
},
```

### Expected results

1. No error for that dart file.
2. Android app can compile.

### Before change:
Fails to build for android when you run debug through **VS Code**.
Android app does not compile.

Builds fine with cli:
```console
flutter run \ --flavor development \ --target lib/main/main_development.dart
```
Android app compiles.

### After change:
The app does not compile due to dependencies version failing when running flutter pub-get:
**intl: ^0.17.0**

I fixed that by bumping both daily globe pubpec.yaml and packages:app_ui/pubspec.yaml to:
**intl: ^0.18.0**

Then it fails to compile again:
```console
Launching lib/main/main_development.dart on sdk gphone64 x86 64 in debug mode...
main_development.dart:1
Warning: Mapping new ns http://schemas.android.com/repository/android/common/02 to old ns http://schemas.android.com/repository/android/common/01
Warning: Mapping new ns http://schemas.android.com/repository/android/generic/02 to old ns http://schemas.android.com/repository/android/generic/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/02 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/addon2/03 to old ns http://schemas.android.com/sdk/android/repo/addon2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/02 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/repository2/03 to old ns http://schemas.android.com/sdk/android/repo/repository2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/03 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
Warning: Mapping new ns http://schemas.android.com/sdk/android/repo/sys-img2/02 to old ns http://schemas.android.com/sdk/android/repo/sys-img2/01
: Error: The argument type 'void Function(String?, RenderContext, Map<String, String>)' can't be assigned to the parameter type 'void Function(String?, RenderContext, Map<String, String>, Element?)?'.
html.dart:24
- 'RenderContext' is from 'package:flutter_html/html_parser.dart' ('../../../../../AppData/Local/Pub/Cache/hosted/pub.dev/flutter_html-3.0.0-alpha.6/lib/html_parser.dart').
html_parser.dart:1
 - 'Map' is from 'dart:core'.
- 'Element' is from 'package:html/dom.dart' ('../../../../../AppData/Local/Pub/Cache/hosted/pub.dev/html-0.15.1/lib/dom.dart').
dom.dart:1
        onLinkTap: (url, context, attributes) {
Target kernel_snapshot failed: Exception
2

FAILURE: Build failed with an exception.

* Where:
Script 'C:\Src\flutter\packages\flutter_tools\gradle\flutter.gradle' line: 1159

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDevelopmentDebug'.
> Process 'command 'C:\Src\flutter\bin\flutter.bat'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 16s
Exception: Gradle task assembleDevelopmentDebug failed with exit code 1
```

### I need some guidance please:
- I have added sample code for the change but it breaks.
- I am also unsure as to why the widget is also being passed as a parameters.
- Please show me the correct way of fixing the errors if my fixes are wrong. I can do a PR if allowed.

Thank you.

### Code sample

<details><summary>packages:new_blocks_ui/lib/src/html.dart line 24 - 29</summary>

```dart
onLinkTap: (url, context, attributes, element) {
  if (url == null) return;
  final uri = Uri.tryParse(url);
  if (uri == null) return;
  launchUrl(uri).ignore();
},     
```

</details>

I fixed it by removing the *element* param for onLinkTap():
<details><summary>packages:new_blocks_ui/lib/src/html.dart line 24 - 29</summary>

```dart
onLinkTap: (url, context, attributes) {
  if (url == null) return;
  final uri = Uri.tryParse(url);
  if (uri == null) return;
  launchUrl(uri).ignore();
},     
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[√] Flutter (Channel stable, 3.10.0, on Microsoft Windows [Version 10.0.19044.2965], locale
    en-NA)
    • Flutter version 3.10.0 on channel stable at C:\Src\flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 84a1e904f4 (8 days ago), 2023-05-09 07:41:44 -0700
    • Engine revision d44b5a94c9
    • Dart version 3.0.0
    • DevTools version 2.23.1

[√] Windows Version (Installed version of Windows is version 10 or higher)

[√] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at C:\Users\TylerByte\AppData\Local\Android\sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jbr\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.3.3)
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.3.32825.248
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2022.1)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-9505619)

[√] VS Code (version 1.78.2)
    • VS Code at C:\Users\TylerByte\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.54.0

[√] Connected device (4 available)
    • sdk gphone64 x86 64 (mobile) • emulator-5554 • android-x64    • Android 13 (API 33)      
      (emulator)
    • Windows (desktop)            • windows       • windows-x64    • Microsoft Windows
      [Version 10.0.19044.2965]
    • Chrome (web)                 • chrome        • web-javascript • Google Chrome
      113.0.5672.93
    • Edge (web)                   • edge          • web-javascript • Microsoft Edge
      112.0.1722.39

[√] Network resources
    • All expected network resources are available.

• No issues found!
```

</details>

**Comment by felangel** [comment]
cc @marwfair @kaiceyd

**Comment by dam-ease** [comment]
Hi @TylerByte666. Thanks for filling this issue. I am able to reproduce on the latest `stable` and `master` channels.

**Comment by zoeyfan** [comment]
Hey Kevin, would you please pass this to the right team member to look at it? Or would you please let me know who I can pass this to?

**Comment by doodatales** [comment]
Still having this problem. Could write a hack around it, but would rather not.

**Comment by TylerByte666** [comment]
Hi, still having this issue not sure why this was closed?

**Comment by valentinallavayol** [comment]
> Hi, still having this issue not sure why this was closed?

Hey @TylerByte666 ! This issue was resolved on version 1.0.5 of the mason brick of the toolkit. In order to upgrade your project to the latest version (1.0.6) please follow the steps on [this](https://flutter.github.io/news_toolkit/project_configuration/upgrading/) documentation page.

**Comment by TylerByte666** [comment]
Thank you so much, it was an ID10T error from my side :P
